variables.tf

```diff
@@ -61,17 +61,86 @@ variable "deployment" {
   nullable    = false
 }
 
+variable "dynamic_throttling_enabled" {
+  type        = bool
+  default     = null
+  description = "Determines whether or not dynamic throttling is enabled. If set to true, dynamic throttling will be enabled. If set to false, dynamic throttling will not be enabled."
+}
+
 variable "environment" {
   type        = string
   default     = ""
   description = "Environment of the application. A corresponding tag would be created on the created resources if `var.default_tags_enabled` is `true`."
 }
 
-# tflint-ignore: terraform_unused_declarations
-variable "network_acls_default_action" {
-  type        = string
-  default     = "Deny"
-  description = "Azure OpenAI account network ACLs default action."
+variable "customer_managed_key" {
+  type = object({
+    key_vault_key_id   = string
+    identity_client_id = optional(string)
+  })
+  default     = null
+  description = <<-DESCRIPTION
+    type = object({
+      key_vault_key_id   = (Required) The ID of the Key Vault Key which should be used to Encrypt the data in this OpenAI Account.
+      identity_client_id = (Optional) The Client ID of the User Assigned Identity that has access to the key. This property only needs to be specified when there're multiple identities attached to the OpenAI Account.
+    })
+  DESCRIPTION
+  sensitive   = true
+}
+
+variable "fqdns" {
+  type        = list(string)
+  default     = []
+  description = "List of FQDNs allowed for the Cognitive Account."
+}
+
+
+variable "identity" {
+  type = object({
+    identity_ids = optional(list(string))
+    type         = string
+  })
+  default     = null
+  description = <<-DESCRIPTION
+    type = object({
+      type         = (Required) The type of the Identity. Possible values are 'SystemAssigned', 'UserAssigned', 'SystemAssigned, UserAssigned'.
+      identity_ids = (Optional) Specifies a list of User Assigned Managed Identity IDs to be assigned to this OpenAI Account.
+    })
+  DESCRIPTION
+}
+
+variable "local_auth_enabled" {
+  type        = bool
+  default     = true
+  description = "Whether local authentication methods is enabled for the Cognitive Account. Defaults to true."
+}
+
+variable "network_acls" {
+  type = object({
+    default_action = string
+    ip_rules       = optional(list(string))
+    virtual_network_rules = optional(object({
```

Comment:
According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.56.0/internal/services/cognitive/cognitive_account_resource.go#L221-L225), the type here should be `optional(set(object{...}), [])